### DISCUSSION TITLE
Scarce Robotics v0.1 (Orchard Inspector)

### DISCUSSION THREAD

**Discussion by stefanesa** [polymathrobotics/caladan_examples#15] [Show and tell]
There are a lot of different autonomy examples in farms, one that has come up time and time again is a robot to drive inbetween rows at an orchard.  I just did a quick pass at driving through the orchard and I thought I'd share a couple things I saw.

But first, my result: 
![image](https://user-images.githubusercontent.com/1851242/180632058-d41590d6-aeb7-4018-aa14-66042c7cb345.png)

To issue these commands I modified a previously build gSheet for coordinate creation.  You can find a copy of it here: https://docs.google.com/spreadsheets/d/1aw2NUHz1xktq0kkg5eOahh8pzMBg5reRKUU2VL_H_l8/edit?usp=sharing

In a previous tillage example, I had my rows end in poses that would face their next row.  Unfortunately, that led to some weird behaviors as the tractor would try to reposition itself at the end of it's row (as opposed to doing a gentle turn, which was my intention).  

For this example, I thought I would solve this problem by having "half Columns" (having the same latitude as the previous spot, but a longitude halfway between the two Columns) inbetween each Orchard Row that served to pre-position my tractor in a good spot for the next column.  Unfortunately that just led to a new set of funky behaviors because, as I learned, our autonomy isn't yet configured to drive in rounded squares.

Next time I'll have my intermediate waypoint a meter or two out from the end of each row, which will hopefully smooth it out.  Also, my south-side latitude was probably too close to the orchard for a smooth behavior.

**Comment by alexanderjyuen** [comment]
Hey Stefan, so this is "normal" for our local planner. Normal in the sense that with a scarce number of points, the local planner just kind of makes it up as it goes a long. It tries to find the happy medium between a bunch of competing factors and the resulting path can sometimes turn out kind of odd.

This is where something like fields2cover remedies this. By providing more informaiton about the geometry of the turns at the the ends of the tilling, you're effectively constraining the vehicle to a path that you want
